### ISSUE TITLE
Admin user creation doesn't work in my vagrant

### ISSUE THREAD

**Issue by flozano** [basho-labs/vagrant-riak-cs-cluster#9]
I got the following when trying to use this box (vagrant 1.1.5, MacOS X 10.8):

```
$ RIAK_CS_CREATE_ADMIN_USER=1 vagrant up
Bringing machine 'riak1' up with 'virtualbox' provider...
[riak1] Importing base box 'opscode-ubuntu-12.04'...
[riak1] Matching MAC address for NAT networking...
[riak1] Setting the name of the VM...
[riak1] Clearing any previously set forwarded ports...
[riak1] Creating shared folders metadata...
[riak1] Clearing any previously set network interfaces...
[riak1] Preparing network interfaces based on configuration...
[riak1] Forwarding ports...
[riak1] -- 22 => 2222 (adapter 1)
[riak1] -- 8080 => 8080 (adapter 1)
[riak1] -- 8085 => 8085 (adapter 1)
[riak1] -- 8087 => 8087 (adapter 1)
[riak1] Running any VM customizations...
[riak1] Booting VM...
[riak1] Waiting for VM to boot. This can take a few minutes.
[riak1] VM booted and ready for use!
[riak1] Setting hostname...
[riak1] Configuring and enabling network interfaces...
[riak1] Mounting shared folders...
[riak1] -- /vagrant
[riak1] -- /tmp/vagrant-chef-1/lib
[riak1] -- /tmp/vagrant-chef-1/chef-solo-2/roles
[riak1] -- /tmp/vagrant-chef-1/chef-solo-1/cookbooks
[riak1] Running provisioner: VagrantPlugins::Shell::Provisioner...
stdin: is not a tty
dpkg-preconfigure: unable to re-open stdin: No such file or directory
Selecting previously unselected package liberror-perl.
(Reading database ... 
60953 files and directories currently installed.)
Unpacking liberror-perl (from .../liberror-perl_0.17-1_all.deb) ...
Selecting previously unselected package git-man.
Unpacking git-man (from .../git-man_1%3a1.7.9.5-1_all.deb) ...
Selecting previously unselected package git.
Unpacking git (from .../git_1%3a1.7.9.5-1_amd64.deb) ...
Processing triggers for man-db ...
Setting up liberror-perl (0.17-1) ...
Setting up git-man (1:1.7.9.5-1) ...
Setting up git (1:1.7.9.5-1) ...
Installing berkshelf
Successfully installed timers-1.1.0
Successfully installed celluloid-0.13.0
Successfully installed i18n-0.6.1
Successfully installed multi_json-1.7.2
Successfully installed activesupport-3.2.13
Successfully installed multipart-post-1.2.0
Successfully installed faraday-0.8.7
Successfully installed hashie-2.0.3
Successfully installed chozo-0.6.1
Successfully installed addressable-2.3.3
Successfully installed solve-0.4.2
Successfully installed net-http-persistent-2.8
Successfully installed retryable-1.3.2
Successfully installed ridley-0.9.0
Successfully installed minitar-0.5.4
Successfully installed thor-0.18.1
Successfully installed berkshelf-1.4.0
17 gems installed
Berkshelf: Installing cookbooks
Installing apt (1.9.2) from site: 'http://cookbooks.opscode.com/api/v1/cookbooks'
Installing yum (2.2.0) from site: 'http://cookbooks.opscode.com/api/v1/cookbooks'
Installing ntp (1.3.2) from site: 'http://cookbooks.opscode.com/api/v1/cookbooks'
Installing nmap (0.1.0) from site: 'http://cookbooks.opscode.com/api/v1/cookbooks'
Installing htop (1.0.0) from site: 'http://cookbooks.opscode.com/api/v1/cookbooks'
Installing openssh (1.1.4) from site: 'http://cookbooks.opscode.com/api/v1/cookbooks'
Installing sudo (2.0.4) from site: 'http://cookbooks.opscode.com/api/v1/cookbooks'
Installing riak-cs (1.3.1) from github: 'basho/riak-cs-chef-cookbook' with branch: '5cb5de635d118b67569094f8ec82ed621dcf878c' over protocol: 'git'
Installing riak (1.3.1) from github: 'basho/riak-chef-cookbook' with branch: '1ee315af8ed49963dac187e4a4ab5f59f4bb7880' over protocol: 'git'
Installing riak-cs-create-admin-user (0.1.0) from github: 'hectcastro/chef-riak-cs-create-admin-user' with branch: '54b1592c214b0002567488ec0ce0e59ce2d465b0' over protocol: 'git'
Installing build-essential (1.3.4) from site: 'http://cookbooks.opscode.com/api/v1/cookbooks'
Installing erlang (1.2.0) from site: 'http://cookbooks.opscode.com/api/v1/cookbooks'
[riak1] Running provisioner: VagrantPlugins::Chef::Provisioner::ChefSolo...
Generating chef JSON and uploading...
Running chef-solo...
stdin: is not a tty
[2013-04-13T14:56:25+00:00] INFO: *** Chef 11.2.0 ***
[2013-04-13T14:56:25+00:00] INFO: Setting the run_list to ["recipe[apt]", "role[base]", "role[riak]", "role[stanchion]", "role[riak_cs]", "recipe[riak-cs-create-admin-user]"] from JSON
[2013-04-13T14:56:25+00:00] INFO: Run List is [recipe[apt], role[base], role[riak], role[stanchion], role[riak_cs], recipe[riak-cs-create-admin-user]]
[2013-04-13T14:56:25+00:00] INFO: Run List expands to [apt, ntp, nmap, htop, openssh, sudo, riak, riak-cs::stanchion, riak-cs, riak-cs-create-admin-user]
[2013-04-13T14:56:25+00:00] INFO: Starting Chef Run for riak1
[2013-04-13T14:56:25+00:00] INFO: Running start handlers
[2013-04-13T14:56:25+00:00] INFO: Start handlers complete.
[2013-04-13T14:56:26+00:00] WARN: found a directory d20130413-4069-hvfuhp in the cookbook path, but it contains no cookbook files. skipping.
[2013-04-13T14:56:26+00:00] INFO: AptPreference light-weight provider already initialized -- overriding!
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.2.0/lib/chef/provider/lwrp_base.rb:97: warning: already initialized constant AptPreference
[2013-04-13T14:56:26+00:00] INFO: AptRepository light-weight provider already initialized -- overriding!
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.2.0/lib/chef/provider/lwrp_base.rb:97: warning: already initialized constant AptRepository
[2013-04-13T14:56:26+00:00] INFO: AptPreference light-weight resource already initialized -- overriding!
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.2.0/lib/chef/resource/lwrp_base.rb:51: warning: already initialized constant AptPreference
[2013-04-13T14:56:26+00:00] INFO: AptRepository light-weight resource already initialized -- overriding!
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.2.0/lib/chef/resource/lwrp_base.rb:51: warning: already initialized constant AptRepository
[2013-04-13T14:56:26+00:00] INFO: Overriding duplicate definition file_ulimit, new definition found in /tmp/vagrant-chef-1/chef-solo-1/cookbooks/riak-cs/definitions/file_ulimit.rb
[2013-04-13T14:56:26+00:00] INFO: Processing execute[apt-get-update] action run (apt::default line 22)
[2013-04-13T14:56:34+00:00] INFO: execute[apt-get-update] ran successfully
[2013-04-13T14:56:34+00:00] INFO: Processing execute[apt-get update] action nothing (apt::default line 29)
[2013-04-13T14:56:34+00:00] INFO: Processing execute[apt-get autoremove] action nothing (apt::default line 36)
[2013-04-13T14:56:34+00:00] INFO: Processing execute[apt-get autoclean] action nothing (apt::default line 42)
[2013-04-13T14:56:34+00:00] INFO: Processing package[update-notifier-common] action install (apt::default line 48)
[2013-04-13T14:56:36+00:00] INFO: package[update-notifier-common] sending run action to execute[apt-get-update] (immediate)
[2013-04-13T14:56:36+00:00] INFO: Processing execute[apt-get-update] action run (apt::default line 22)
[2013-04-13T14:56:40+00:00] INFO: execute[apt-get-update] ran successfully
[2013-04-13T14:56:40+00:00] INFO: Processing execute[apt-get-update-periodic] action run (apt::default line 52)
[2013-04-13T14:56:40+00:00] INFO: Processing directory[/var/cache/local] action create (apt::default line 62)
[2013-04-13T14:56:40+00:00] INFO: directory[/var/cache/local] created directory /var/cache/local
[2013-04-13T14:56:40+00:00] INFO: directory[/var/cache/local] owner changed to 0
[2013-04-13T14:56:40+00:00] INFO: directory[/var/cache/local] group changed to 0
[2013-04-13T14:56:40+00:00] INFO: directory[/var/cache/local] mode changed to 755
[2013-04-13T14:56:40+00:00] INFO: Processing directory[/var/cache/local/preseeding] action create (apt::default line 62)
[2013-04-13T14:56:40+00:00] INFO: directory[/var/cache/local/preseeding] created directory /var/cache/local/preseeding
[2013-04-13T14:56:40+00:00] INFO: directory[/var/cache/local/preseeding] owner changed to 0
[2013-04-13T14:56:40+00:00] INFO: directory[/var/cache/local/preseeding] group changed to 0
[2013-04-13T14:56:40+00:00] INFO: directory[/var/cache/local/preseeding] mode changed to 755
[2013-04-13T14:56:40+00:00] INFO: Processing package[ntp] action install (ntp::default line 21)
[2013-04-13T14:56:43+00:00] INFO: Processing package[ntpdate] action install (ntp::default line 21)
[2013-04-13T14:56:43+00:00] INFO: Processing directory[/var/lib/ntp] action create (ntp::default line 26)
[2013-04-13T14:56:43+00:00] INFO: Processing directory[/var/log/ntpstats/] action create (ntp::default line 26)
[2013-04-13T14:56:43+00:00] INFO: Processing service[ntp] action enable (ntp::default line 33)
[2013-04-13T14:56:43+00:00] INFO: Processing service[ntp] action start (ntp::default line 33)
[2013-04-13T14:56:43+00:00] INFO: Processing cookbook_file[/etc/ntp.leapseconds] action create (ntp::default line 38)
[2013-04-13T14:56:43+00:00] INFO: cookbook_file[/etc/ntp.leapseconds] owner changed to 0
[2013-04-13T14:56:43+00:00] INFO: cookbook_file[/etc/ntp.leapseconds] group changed to 0
[2013-04-13T14:56:43+00:00] INFO: cookbook_file[/etc/ntp.leapseconds] mode changed to 644
[2013-04-13T14:56:43+00:00] INFO: cookbook_file[/etc/ntp.leapseconds] created file /etc/ntp.leapseconds
[2013-04-13T14:56:43+00:00] INFO: Processing template[/etc/ntp.conf] action create (ntp::default line 44)
[2013-04-13T14:56:44+00:00] INFO: template[/etc/ntp.conf] backed up to /var/chef/backup/etc/ntp.conf.chef-20130413145644
[2013-04-13T14:56:44+00:00] INFO: template[/etc/ntp.conf] updated content
[2013-04-13T14:56:44+00:00] INFO: template[/etc/ntp.conf] owner changed to 0
[2013-04-13T14:56:44+00:00] INFO: template[/etc/ntp.conf] group changed to 0
[2013-04-13T14:56:44+00:00] INFO: template[/etc/ntp.conf] mode changed to 644
[2013-04-13T14:56:44+00:00] INFO: Processing package[nmap] action install (nmap::default line 11)
[2013-04-13T14:56:49+00:00] INFO: Processing package[htop] action install (htop::default line 8)
[2013-04-13T14:56:50+00:00] INFO: Processing package[openssh-client] action install (openssh::default line 27)
[2013-04-13T14:56:50+00:00] INFO: Processing package[openssh-server] action install (openssh::default line 27)
[2013-04-13T14:56:50+00:00] INFO: Processing service[ssh] action enable (openssh::default line 30)
[2013-04-13T14:56:50+00:00] INFO: service[ssh] enabled
[2013-04-13T14:56:50+00:00] INFO: Processing service[ssh] action start (openssh::default line 30)
[2013-04-13T14:56:50+00:00] INFO: Processing template[/etc/ssh/ssh_config] action create (openssh::default line 48)
[2013-04-13T14:56:50+00:00] INFO: template[/etc/ssh/ssh_config] backed up to /var/chef/backup/etc/ssh/ssh_config.chef-20130413145650
[2013-04-13T14:56:50+00:00] INFO: template[/etc/ssh/ssh_config] updated content
[2013-04-13T14:56:50+00:00] INFO: template[/etc/ssh/ssh_config] owner changed to 0
[2013-04-13T14:56:50+00:00] INFO: template[/etc/ssh/ssh_config] group changed to 0
[2013-04-13T14:56:50+00:00] INFO: template[/etc/ssh/ssh_config] mode changed to 644
[2013-04-13T14:56:50+00:00] INFO: Processing template[/etc/ssh/sshd_config] action create (openssh::default line 66)
[2013-04-13T14:56:51+00:00] INFO: template[/etc/ssh/sshd_config] backed up to /var/chef/backup/etc/ssh/sshd_config.chef-20130413145651
[2013-04-13T14:56:51+00:00] INFO: template[/etc/ssh/sshd_config] updated content
[2013-04-13T14:56:51+00:00] INFO: template[/etc/ssh/sshd_config] owner changed to 0
[2013-04-13T14:56:51+00:00] INFO: template[/etc/ssh/sshd_config] group changed to 0
[2013-04-13T14:56:51+00:00] INFO: template[/etc/ssh/sshd_config] mode changed to 644
[2013-04-13T14:56:51+00:00] INFO: Processing package[sudo] action install (sudo::default line 20)
[2013-04-13T14:56:51+00:00] INFO: Processing template[/etc/sudoers] action create (sudo::default line 41)
[2013-04-13T14:56:51+00:00] INFO: template[/etc/sudoers] backed up to /var/chef/backup/etc/sudoers.chef-20130413145651
[2013-04-13T14:56:51+00:00] INFO: template[/etc/sudoers] updated content
[2013-04-13T14:56:51+00:00] INFO: template[/etc/sudoers] owner changed to 0
[2013-04-13T14:56:51+00:00] INFO: template[/etc/sudoers] group changed to 0
[2013-04-13T14:56:51+00:00] INFO: template[/etc/sudoers] mode changed to 440
[2013-04-13T14:56:51+00:00] INFO: Processing apt_repository[basho] action add (riak::default line 49)
[2013-04-13T14:56:51+00:00] INFO: Processing remote_file[/tmp/vagrant-chef-1/basho.apt.key] action create (/tmp/vagrant-chef-1/chef-solo-1/cookbooks/apt/providers/repository.rb line 53)
[2013-04-13T14:56:51+00:00] INFO: remote_file[/tmp/vagrant-chef-1/basho.apt.key] updated
[2013-04-13T14:56:51+00:00] INFO: remote_file[/tmp/vagrant-chef-1/basho.apt.key] mode changed to 644
[2013-04-13T14:56:51+00:00] INFO: Processing execute[install-key basho.apt.key] action run (/tmp/vagrant-chef-1/chef-solo-1/cookbooks/apt/providers/repository.rb line 67)
[2013-04-13T14:56:51+00:00] INFO: execute[install-key basho.apt.key] ran successfully
[2013-04-13T14:56:51+00:00] INFO: Processing file[/var/lib/apt/periodic/update-success-stamp] action nothing (/tmp/vagrant-chef-1/chef-solo-1/cookbooks/apt/providers/repository.rb line 100)
[2013-04-13T14:56:51+00:00] INFO: Processing execute[apt-get update] action nothing (/tmp/vagrant-chef-1/chef-solo-1/cookbooks/apt/providers/repository.rb line 104)
[2013-04-13T14:56:51+00:00] INFO: Processing file[/etc/apt/sources.list.d/basho.list] action create (/tmp/vagrant-chef-1/chef-solo-1/cookbooks/apt/providers/repository.rb line 116)
[2013-04-13T14:56:52+00:00] INFO: entered create
[2013-04-13T14:56:52+00:00] INFO: file[/etc/apt/sources.list.d/basho.list] owner changed to 0
[2013-04-13T14:56:52+00:00] INFO: file[/etc/apt/sources.list.d/basho.list] group changed to 0
[2013-04-13T14:56:52+00:00] INFO: file[/etc/apt/sources.list.d/basho.list] mode changed to 644
[2013-04-13T14:56:52+00:00] INFO: file[/etc/apt/sources.list.d/basho.list] created file /etc/apt/sources.list.d/basho.list
[2013-04-13T14:56:52+00:00] INFO: file[/etc/apt/sources.list.d/basho.list] sending delete action to file[/var/lib/apt/periodic/update-success-stamp] (immediate)
[2013-04-13T14:56:52+00:00] INFO: Processing file[/var/lib/apt/periodic/update-success-stamp] action delete (/tmp/vagrant-chef-1/chef-solo-1/cookbooks/apt/providers/repository.rb line 100)
[2013-04-13T14:56:52+00:00] INFO: file[/var/lib/apt/periodic/update-success-stamp] backed up to /var/chef/backup/var/lib/apt/periodic/update-success-stamp.chef-20130413145652
[2013-04-13T14:56:52+00:00] INFO: file[/var/lib/apt/periodic/update-success-stamp] deleted file at /var/lib/apt/periodic/update-success-stamp
[2013-04-13T14:56:52+00:00] INFO: file[/etc/apt/sources.list.d/basho.list] sending run action to execute[apt-get update] (immediate)
[2013-04-13T14:56:52+00:00] INFO: Processing execute[apt-get update] action run (/tmp/vagrant-chef-1/chef-solo-1/cookbooks/apt/providers/repository.rb line 104)
[2013-04-13T14:56:55+00:00] INFO: execute[apt-get update] ran successfully
[2013-04-13T14:56:55+00:00] INFO: Processing package[riak] action install (riak::default line 56)
[2013-04-13T14:57:14+00:00] INFO: Processing file[/etc/riak/app.config] action create (riak::default line 96)
[2013-04-13T14:57:14+00:00] INFO: file[/etc/riak/app.config] backed up to /var/chef/backup/etc/riak/app.config.chef-20130413145714
[2013-04-13T14:57:14+00:00] INFO: file[/etc/riak/app.config] contents updated
[2013-04-13T14:57:14+00:00] INFO: Processing file[/etc/riak/vm.args] action create (riak::default line 103)
[2013-04-13T14:57:14+00:00] INFO: file[/etc/riak/vm.args] backed up to /var/chef/backup/etc/riak/vm.args.chef-20130413145714
[2013-04-13T14:57:14+00:00] INFO: file[/etc/riak/vm.args] contents updated
[2013-04-13T14:57:14+00:00] INFO: file[/etc/riak/vm.args] not queuing delayed action restart on service[riak] (delayed), as it's already been queued
[2013-04-13T14:57:14+00:00] INFO: Processing service[riak] action enable (riak::default line 120)
[2013-04-13T14:57:14+00:00] INFO: Processing service[riak] action start (riak::default line 120)
[2013-04-13T14:57:33+00:00] INFO: service[riak] started
[2013-04-13T14:57:33+00:00] INFO: Processing package[libssl0.9.8] action install (riak-cs::stanchion line 30)
[2013-04-13T14:57:36+00:00] INFO: Processing group[stanchion] action create (riak-cs::stanchion line 51)
[2013-04-13T14:57:36+00:00] INFO: group[stanchion] created
[2013-04-13T14:57:36+00:00] INFO: Processing user[stanchion] action create (riak-cs::stanchion line 53)
[2013-04-13T14:57:36+00:00] INFO: user[stanchion] created
[2013-04-13T14:57:36+00:00] INFO: Processing remote_file[/tmp/vagrant-chef-1/stanchion_1.3.1-1_amd64.deb] action create (riak-cs::stanchion line 60)
[2013-04-13T14:57:46+00:00] INFO: remote_file[/tmp/vagrant-chef-1/stanchion_1.3.1-1_amd64.deb] updated
[2013-04-13T14:57:46+00:00] INFO: remote_file[/tmp/vagrant-chef-1/stanchion_1.3.1-1_amd64.deb] owner changed to 0
[2013-04-13T14:57:46+00:00] INFO: remote_file[/tmp/vagrant-chef-1/stanchion_1.3.1-1_amd64.deb] mode changed to 644
[2013-04-13T14:57:46+00:00] INFO: Processing package[stanchion_1.3.1-1_amd64.deb] action install (riak-cs::stanchion line 69)
[2013-04-13T14:57:47+00:00] INFO: Processing directory[/etc/stanchion] action create (riak-cs::stanchion line 83)
[2013-04-13T14:57:47+00:00] INFO: directory[/etc/stanchion] owner changed to 0
[2013-04-13T14:57:47+00:00] INFO: Processing file[/etc/stanchion/app.config] action create (riak-cs::stanchion line 89)
[2013-04-13T14:57:47+00:00] INFO: file[/etc/stanchion/app.config] backed up to /var/chef/backup/etc/stanchion/app.config.chef-20130413145747
[2013-04-13T14:57:47+00:00] INFO: file[/etc/stanchion/app.config] contents updated
[2013-04-13T14:57:47+00:00] INFO: file[/etc/stanchion/app.config] owner changed to 0
[2013-04-13T14:57:47+00:00] INFO: Processing file[/etc/stanchion/vm.args] action create (riak-cs::stanchion line 96)
[2013-04-13T14:57:48+00:00] INFO: file[/etc/stanchion/vm.args] backed up to /var/chef/backup/etc/stanchion/vm.args.chef-20130413145748
[2013-04-13T14:57:48+00:00] INFO: file[/etc/stanchion/vm.args] contents updated
[2013-04-13T14:57:48+00:00] INFO: file[/etc/stanchion/vm.args] owner changed to 0
[2013-04-13T14:57:48+00:00] INFO: file[/etc/stanchion/vm.args] not queuing delayed action restart on service[stanchion] (delayed), as it's already been queued
[2013-04-13T14:57:48+00:00] INFO: Processing service[stanchion] action enable (riak-cs::stanchion line 111)
[2013-04-13T14:57:48+00:00] INFO: Processing service[stanchion] action start (riak-cs::stanchion line 111)
[2013-04-13T14:58:05+00:00] INFO: service[stanchion] started
[2013-04-13T14:58:05+00:00] INFO: Processing group[riak] action create (riak-cs::default line 50)
[2013-04-13T14:58:05+00:00] INFO: Processing user[riakcs] action create (riak-cs::default line 52)
[2013-04-13T14:58:05+00:00] INFO: user[riakcs] created
[2013-04-13T14:58:05+00:00] INFO: Processing remote_file[/tmp/vagrant-chef-1/riak-cs_1.3.1-1_amd64.deb] action create (riak-cs::default line 59)
[2013-04-13T14:58:20+00:00] INFO: remote_file[/tmp/vagrant-chef-1/riak-cs_1.3.1-1_amd64.deb] updated
[2013-04-13T14:58:20+00:00] INFO: remote_file[/tmp/vagrant-chef-1/riak-cs_1.3.1-1_amd64.deb] owner changed to 0
[2013-04-13T14:58:20+00:00] INFO: remote_file[/tmp/vagrant-chef-1/riak-cs_1.3.1-1_amd64.deb] mode changed to 644
[2013-04-13T14:58:21+00:00] INFO: Processing directory[/etc/riak-cs] action create (riak-cs::default line 66)
[2013-04-13T14:58:21+00:00] INFO: directory[/etc/riak-cs] created directory /etc/riak-cs
[2013-04-13T14:58:21+00:00] INFO: directory[/etc/riak-cs] owner changed to 0
[2013-04-13T14:58:21+00:00] INFO: directory[/etc/riak-cs] mode changed to 755
[2013-04-13T14:58:21+00:00] INFO: Processing package[riak-cs_1.3.1-1_amd64.deb] action install (riak-cs::default line 72)
[2013-04-13T14:58:22+00:00] INFO: Processing file[/etc/riak-cs/app.config] action create (riak-cs::default line 84)
[2013-04-13T14:58:22+00:00] INFO: file[/etc/riak-cs/app.config] backed up to /var/chef/backup/etc/riak-cs/app.config.chef-20130413145822
[2013-04-13T14:58:22+00:00] INFO: file[/etc/riak-cs/app.config] contents updated
[2013-04-13T14:58:22+00:00] INFO: file[/etc/riak-cs/app.config] owner changed to 0
[2013-04-13T14:58:22+00:00] INFO: Processing file[/etc/riak-cs/vm.args] action create (riak-cs::default line 91)
[2013-04-13T14:58:22+00:00] INFO: file[/etc/riak-cs/vm.args] backed up to /var/chef/backup/etc/riak-cs/vm.args.chef-20130413145822
[2013-04-13T14:58:22+00:00] INFO: file[/etc/riak-cs/vm.args] contents updated
[2013-04-13T14:58:22+00:00] INFO: file[/etc/riak-cs/vm.args] owner changed to 0
[2013-04-13T14:58:22+00:00] INFO: file[/etc/riak-cs/vm.args] not queuing delayed action restart on service[riak-cs] (delayed), as it's already been queued
[2013-04-13T14:58:22+00:00] INFO: Processing service[riak-cs] action enable (riak-cs::default line 108)
[2013-04-13T14:58:22+00:00] INFO: Processing service[riak-cs] action start (riak-cs::default line 108)
[2013-04-13T14:58:39+00:00] INFO: service[riak-cs] started
[2013-04-13T14:58:39+00:00] INFO: Processing ruby_block[trigger-delayed-restarts] action run (riak-cs-create-admin-user::default line 1)
[2013-04-13T14:58:39+00:00] INFO: ruby_block[trigger-delayed-restarts] called
[2013-04-13T14:58:39+00:00] INFO: ruby_block[trigger-delayed-restarts] sending restart action to service[riak] (immediate)
[2013-04-13T14:58:39+00:00] INFO: Processing service[riak] action restart (riak::default line 120)
[2013-04-13T14:59:01+00:00] INFO: service[riak] restarted
[2013-04-13T14:59:01+00:00] INFO: ruby_block[trigger-delayed-restarts] sending restart action to service[stanchion] (immediate)
[2013-04-13T14:59:01+00:00] INFO: Processing service[stanchion] action restart (riak-cs::stanchion line 111)
[2013-04-13T14:59:18+00:00] INFO: service[stanchion] restarted
[2013-04-13T14:59:18+00:00] INFO: ruby_block[trigger-delayed-restarts] sending restart action to service[riak-cs] (immediate)
[2013-04-13T14:59:18+00:00] INFO: Processing service[riak-cs] action restart (riak-cs::default line 108)
[2013-04-13T14:59:36+00:00] INFO: service[riak-cs] restarted
[2013-04-13T14:59:36+00:00] INFO: Processing ruby_block[create-admin-user] action run (riak-cs-create-admin-user::default line 8)
[2013-04-13T14:59:36+00:00] INFO: Retrying execution of ruby_block[create-admin-user], 4 attempt(s) left
[2013-04-13T14:59:39+00:00] INFO: Retrying execution of ruby_block[create-admin-user], 3 attempt(s) left
[2013-04-13T14:59:42+00:00] INFO: Retrying execution of ruby_block[create-admin-user], 2 attempt(s) left
[2013-04-13T14:59:45+00:00] INFO: Retrying execution of ruby_block[create-admin-user], 1 attempt(s) left
[2013-04-13T14:59:48+00:00] INFO: Retrying execution of ruby_block[create-admin-user], 0 attempt(s) left


================================================================================

Error executing action `run` on resource 'ruby_block[create-admin-user]'

================================================================================




Errno::ECONNREFUSED

-------------------

Connection refused - connect(2)




Cookbook Trace:

---------------

/tmp/vagrant-chef-1/chef-solo-1/cookbooks/riak-cs-create-admin-user/libraries/riak_cs_create_admin_user.rb:16:in `create_admin_user'
/tmp/vagrant-chef-1/chef-solo-1/cookbooks/riak-cs-create-admin-user/recipes/default.rb:10:in `block (2 levels) in from_file'


Resource Declaration:
---------------------
# In /tmp/vagrant-chef-1/chef-solo-1/cookbooks/riak-cs-create-admin-user/recipes/default.rb

  8: ruby_block "create-admin-user" do
  9:   block do
 10:     admin_key, admin_secret = RiakCS.create_admin_user(
 11:       node["riak_cs_create_admin_user"]["name"],
 12:       node["riak_cs_create_admin_user"]["email"],
 13:       node["ipaddress"]
 14:     )
 15: 
 16:     Chef::Log.info "Riak CS Key: #{admin_key}"
 17:     Chef::Log.info "Riak CS Secret: #{admin_secret}"
 18: 
 19:     riak_cs_config = node["riak_cs"]["config"].to_hash
 20:     riak_cs_config = riak_cs_config.merge(
 21:       "riak_cs" => riak_cs_config["riak_cs"].merge(
 22:         "admin_key" => admin_key.to_erl_string,
 23:         "admin_secret" => admin_secret.to_erl_string,
 24:         "anonymous_user_creation" => false
 25:       )
 26:     )
 27:     riak_cs_file = resources(:file => "#{node['riak_cs']['package']['config_dir']}/app.config")
 28:     riak_cs_file.content Eth::Config.new(riak_cs_config).pp
 29: 
 30:     stanchion_config = node["stanchion"]["config"].to_hash
 31:     stanchion_config = stanchion_config.merge(
 32:       "stanchion" => stanchion_config["stanchion"].merge(
 33:         "admin_key" => admin_key.to_erl_string,
 34:         "admin_secret" => admin_secret.to_erl_string
 35:       )
 36:     )
 37:     stanchion_file = resources(:file => "#{node['stanchion']['package']['config_dir']}/app.config")
 38:     stanchion_file.content Eth::Config.new(stanchion_config).pp
 39: 
 40:     node.set["riak_cs"]["config"]["riak_cs"]["anonymous_user_creation"] = false
 41:   end
 42: 
 43:   retries 5
 44:   retry_delay 3
 45:   notifies :create, "file[#{node['stanchion']['package']['config_dir']}/app.config]", :immediately
 46:   notifies :create, "file[#{node['riak_cs']['package']['config_dir']}/app.config]", :immediately
 47:   only_if { node["riak_cs"]["config"]["riak_cs"]["anonymous_user_creation"] }
 48: end



Compiled Resource:
------------------
# Declared in /tmp/vagrant-chef-1/chef-solo-1/cookbooks/riak-cs-create-admin-user/recipes/default.rb:8:in `from_file'

ruby_block("create-admin-user") do
  action "run"
  retries 0
  retry_delay 3
  block_name "create-admin-user"
  cookbook_name :"riak-cs-create-admin-user"
  recipe_name "default"
  block #<Proc:0x00000001f93cf0@/tmp/vagrant-chef-1/chef-solo-1/cookbooks/riak-cs-create-admin-user/recipes/default.rb:9>
  only_if { #code block }
end



[2013-04-13T14:59:51+00:00] INFO: Running queued delayed notifications before re-raising exception
[2013-04-13T14:59:51+00:00] INFO: template[/etc/ntp.conf] sending restart action to service[ntp] (delayed)
[2013-04-13T14:59:51+00:00] INFO: Processing service[ntp] action restart (ntp::default line 33)
[2013-04-13T14:59:53+00:00] INFO: service[ntp] restarted
[2013-04-13T14:59:53+00:00] INFO: template[/etc/ssh/sshd_config] sending restart action to service[ssh] (delayed)
[2013-04-13T14:59:53+00:00] INFO: Processing service[ssh] action restart (openssh::default line 30)
[2013-04-13T14:59:53+00:00] INFO: service[ssh] restarted
[2013-04-13T14:59:53+00:00] INFO: file[/etc/riak/app.config] sending restart action to service[riak] (delayed)
[2013-04-13T14:59:53+00:00] INFO: Processing service[riak] action restart (riak::default line 120)
[2013-04-13T15:00:14+00:00] INFO: service[riak] restarted
[2013-04-13T15:00:14+00:00] INFO: file[/etc/stanchion/app.config] sending restart action to service[stanchion] (delayed)
[2013-04-13T15:00:14+00:00] INFO: Processing service[stanchion] action restart (riak-cs::stanchion line 111)
[2013-04-13T15:00:32+00:00] INFO: service[stanchion] restarted
[2013-04-13T15:00:32+00:00] INFO: file[/etc/riak-cs/app.config] sending restart action to service[riak-cs] (delayed)
[2013-04-13T15:00:32+00:00] INFO: Processing service[riak-cs] action restart (riak-cs::default line 108)
[2013-04-13T15:00:49+00:00] INFO: service[riak-cs] restarted
[2013-04-13T15:00:49+00:00] ERROR: Running exception handlers
[2013-04-13T15:00:49+00:00] ERROR: Exception handlers complete
[2013-04-13T15:00:49+00:00] FATAL: Stacktrace dumped to /tmp/vagrant-chef-1/chef-stacktrace.out
[2013-04-13T15:00:49+00:00] FATAL: Errno::ECONNREFUSED: ruby_block[create-admin-user] (riak-cs-create-admin-user::default line 8) had an error: Errno::ECONNREFUSED: Connection refused - connect(2)
Chef never successfully completed! Any errors should be visible in the
output above. Please fix your recipes so that they properly complete.
```

**Comment by hectcastro** [comment]
At the moment `riak-cs-chef-cookbook` is directed at version `1.3.1` and the `riak-chef-cookbook` at `1.3.0`. Once this [pull request](https://github.com/basho/riak-chef-cookbook/pull/59) is merged, the admin user should create successfully.

In the mean time, a workaround is to point Berkshelf at the pull request branch:

``` diff
diff --git a/Berksfile b/Berksfile
index f2f83f7..6d932f4 100644
--- a/Berksfile
+++ b/Berksfile
@@ -9,5 +9,5 @@ cookbook "openssh"
 cookbook "sudo"

 cookbook "riak-cs", github: "basho/riak-cs-chef-cookbook"
-cookbook "riak", github: "basho/riak-chef-cookbook"
+cookbook "riak", github: "basho/riak-chef-cookbook", branch: "riak-cs-1.3.1"
 cookbook "riak-cs-create-admin-user", github: "hectcastro/chef-riak-cs-create-admin-user"
```

Please reopen this ticket if that does not resolve your issue, and apologies for the confusion.

**Comment by flozano** [comment]
Wow, that was fast! 
Thanks a lot, it worked perfectly.

Now I'm blocked by another issue but this is a riak-CS specific one: https://github.com/basho/riak_cs/issues/535 and it's nothing related to vagrant

**Comment by hectcastro** [comment]
`1.3.1` support for Riak CS has been merged into the Riak cookbook. The workaround should no longer be needed.
